includes/template-functions.php

```diff
@@ -925,3 +925,26 @@ function edd_add_download_post_classes( $classes, $class = '', $post_id = false
 	return $classes;
 }
 add_filter( 'post_class', 'edd_add_download_post_classes', 20, 3 );
+
+/**
+ * Adds Download product price to oembed display
+ *
+ * @since 2.6
+ * @return void
+ */
+function edd_add_oembed_price() {
+
+	if( 'download' !== get_post_type( get_the_ID() ) ) {
+		return;
+	}
+
+	$show = ! get_post_meta( get_the_ID(), '_edd_hide_purchase_link', true );
```

Comment:
We need to filter the `$show` variable so that extensions like EDD Hide Downloads can tell the embed not to work if the download is supposed to be hidden.